Framework/PythonInterface/mantid/plots/modest_image/modest_image.py

```diff
@@ -0,0 +1,300 @@
+                raise TypeError("Invalid dimensions for image data")
+
+        self.invalidate_cache()
+
+    def invalidate_cache(self):
+        self._bounds = None
+        self._imcache = None
+        self._rgbacache = None
+        self._oldxslice = None
+        self._oldyslice = None
+        self._sx, self._sy = None, None
+        self._pixel2world_cache = None
+        self._world2pixel_cache = None
+
+    def set_extent(self, extent):
+        self._full_extent = extent
+        self.invalidate_cache()
+        mi.AxesImage.set_extent(self, extent)
+
+    def get_array(self):
+        """Override to return the full-resolution array"""
+        return self._full_res
+
+    @property
+    def _pixel2world(self):
+
+        if self._pixel2world_cache is None:
+
+            # Pre-compute affine transforms to convert between the 'world'
+            # coordinates of the axes (what is shown by the axis labels) to
+            # 'pixel' coordinates in the underlying array.
+
+            extent = self._full_extent
+
+            if extent is None:
+
+                self._pixel2world_cache = IDENTITY_TRANSFORM
+
+            else:
+
+                self._pixel2world_cache = Affine2D()
+
+                self._pixel2world.translate(+0.5, +0.5)
+
+                self._pixel2world.scale((extent[1] - extent[0]) / self._full_res.shape[1],
+                                        (extent[3] - extent[2]) / self._full_res.shape[0])
+
+                self._pixel2world.translate(extent[0], extent[2])
+
+            self._world2pixel_cache = None
+
+        return self._pixel2world_cache
+
+    @property
+    def _world2pixel(self):
+        if self._world2pixel_cache is None:
+            self._world2pixel_cache = self._pixel2world.inverted()
+        return self._world2pixel_cache
+
+    def _scale_to_res(self):
+        """
+        Change self._A and _extent to render an image whose resolution is
+        matched to the eventual rendering.
+        """
+
+        # Find out how we need to slice the array to make sure we match the
+        # resolution of the display. We pass self._world2pixel which matters
+        # for cases where the extent has been set.
+        x0, x1, sx, y0, y1, sy = extract_matched_slices(axes=self.axes,
+                                                        shape=self._full_res.shape,
+                                                        transform=self._world2pixel)
+
+        # Check whether we've already calculated what we need, and if so just
+        # return without doing anything further.
+        if (self._bounds is not None and
+                sx >= self._sx and sy >= self._sy and
+                x0 >= self._bounds[0] and x1 <= self._bounds[1] and
+                y0 >= self._bounds[2] and y1 <= self._bounds[3]):
+            return
+
+        # Slice the array using the slices determined previously to optimally
+        # match the display
+        self._A = self._full_res[y0:y1:sy, x0:x1:sx]
```

Comment:
I think `skimage.transform.resize` here could give additional options (for example the documentation suggests [using Gaussian smoothing when downsampling](http://scikit-image.org/docs/stable/auto_examples/transform/plot_rescale.html)), but I haven't looked to see if it makes 
a significant difference in our colorfill plots or measured if it significantly slows down plotting?